/services/git gives Internal Server Error if an organization cannot be found

```bash
http --form GET https://forge.api.prod-preview.openshift.io/api/services/git/repositories\?organization\=BLAH "Authorization:Bearer $LAUNCHER_TOKEN" X-App:launcher --timeout 100
```

When the organization doesn't exist, an internal server error is thrown

```



2018-03-01 17:21:46,380 ERROR [io.undertow.request] (default task-61) UT005023: Exception handling request to /api/services/git/repositories: org.jboss.resteasy.spi.UnhandledException: java.lang.IllegalStateException: Cannot fetch the repositories for  organization 'gastaldi'
  | at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:77)
  | at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:220)
  | at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:175)
  | at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:418)
  | at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:209)
  | at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:221)
  | at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56)
  | at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51)
  | at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
  | at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85)
  | at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
  | at io.undertow.websockets.jsr.JsrWebSocketFilter.doFilter(JsrWebSocketFilter.java:130)
  | at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
  | at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
  | at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
  | at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
  | at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
  | at org.wildfly.extension.undertow.security.SecurityContextAssociationHandler.handleRequest(SecurityContextAssociationHandler.java:78)
  | at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
  | at org.wildfly.swarm.generated.FaviconErrorHandler.handleRequest(FaviconErrorHandler.java:62)
  | at io.undertow.server.handlers.PathHandler.handleRequest(PathHandler.java:94)
  | at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131)
  | at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
  | at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
  | at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
  | at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
  | at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
  | at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
  | at io.undertow.security.handlers.NotificationReceiverHandler.handleRequest(NotificationReceiverHandler.java:50)
  | at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
  | at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
  | at org.wildfly.extension.undertow.security.jacc.JACCContextIdHandler.handleRequest(JACCContextIdHandler.java:61)
  | at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
  | at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
  | at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:292)
  | at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:81)
  | at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:138)
  | at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:135)
  | at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
  | at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
  | at io.undertow.servlet.api.LegacyThreadSetupActionWrapper$1.call(LegacyThreadSetupActionWrapper.java:44)
  | at io.undertow.servlet.api.LegacyThreadSetupActionWrapper$1.call(LegacyThreadSetupActionWrapper.java:44)
  | at io.undertow.servlet.api.LegacyThreadSetupActionWrapper$1.call(LegacyThreadSetupActionWrapper.java:44)
  | at io.undertow.servlet.api.LegacyThreadSetupActionWrapper$1.call(LegacyThreadSetupActionWrapper.java:44)
  | at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:272)
  | at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:81)
  | at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:104)
  | at io.undertow.server.Connectors.executeRootHandler(Connectors.java:211)
  | at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:809)
  | at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  | at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  | at java.lang.Thread.run(Thread.java:748)
  | Caused by: java.lang.IllegalStateException: Cannot fetch the repositories for  organization 'gastaldi'
  | at io.fabric8.launcher.service.github.impl.KohsukeGitHubServiceImpl.getRepositories(KohsukeGitHubServiceImpl.java:147)
  | at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  | at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  | at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  | at java.lang.reflect.Method.invoke(Method.java:498)
  | at org.jboss.weld.bean.proxy.AbstractBeanInstance.invoke(AbstractBeanInstance.java:38)
  | at org.jboss.weld.bean.proxy.ProxyMethodHandler.invoke(ProxyMethodHandler.java:100)
  | at io.fabric8.launcher.service.git.api.GitService$5120688$Proxy$_$$_WeldClientProxy.getRepositories(Unknown Source)
  | at io.fabric8.launcher.web.endpoints.GitEndpoint.getRepositories(GitEndpoint.java:77)
  | at io.fabric8.launcher.web.endpoints.GitEndpoint$Proxy$_$$_WeldSubclass.getRepositories(Unknown Source)
  | at io.fabric8.launcher.web.endpoints.GitEndpoint$Proxy$_$$_WeldClientProxy.getRepositories(Unknown Source)
  | at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  | at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  | at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  | at java.lang.reflect.Method.invoke(Method.java:498)
  | at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:139)
  | at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:295)
  | at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:249)
  | at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:236)
  | at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:402)
  | ... 48 more
 

```

--- a/services/git-service-impl/src/main/java/io/fabric8/launcher/service/github/impl/KohsukeGitHubServiceImpl.java
+++ b/services/git-service-impl/src/main/java/io/fabric8/launcher/service/github/impl/KohsukeGitHubServiceImpl.java
@@ -136,16 +136,25 @@ public final class KohsukeGitHubServiceImpl extends AbstractGitService implement
 
     @Override
     public List<GitRepository> getRepositories(GitOrganization organization) {
+        GHPerson person;
         try {
-            GHPerson person = organization != null ? delegate.getOrganization(organization.getName()) : delegate.getMyself();
-            return StreamSupport
-                    .stream(person.listRepositories().spliterator(), false)
-                    .map(r -> new KohsukeGitHubRepositoryImpl(r))
-                    .collect(Collectors.toList());
+            if (organization != null) {
+                try {
+                    person = delegate.getOrganization(organization.getName());
+                } catch (FileNotFoundException e) {
+                    throw new IllegalArgumentException("User does not belong to organization '" + organization.getName() + "' or the organization does not exist", e);
+                }
+            } else {
+                person = delegate.getMyself();
+            }
         } catch (IOException e) {
-            String name = organization != null ? " organization '" + organization.getName() + "'" : "this user";
+            String name = organization != null ? "organization '" + organization.getName() + "'" : "this user";
             throw new IllegalStateException("Cannot fetch the repositories for " + name, e);
         }
+        return StreamSupport
+                .stream(person.listRepositories().spliterator(), false)
+                .map(KohsukeGitHubRepositoryImpl::new)
+                .collect(Collectors.toList());
     }
 
     @Override

--- a/web/src/main/java/io/fabric8/launcher/web/providers/IllegalStateExceptionMapper.java
+++ b/web/src/main/java/io/fabric8/launcher/web/providers/IllegalStateExceptionMapper.java
@@ -1,0 +1,28 @@
+package io.fabric8.launcher.web.providers;
+
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import static javax.json.Json.createArrayBuilder;
+import static javax.json.Json.createObjectBuilder;
+
+/**
+ * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
+ */
+@Provider
+public class IllegalStateExceptionMapper implements ExceptionMapper<IllegalStateException> {
+
+    @Override
+    public Response toResponse(IllegalStateException exception) {
+        return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
+                .entity(createArrayBuilder()
+                                .add(createObjectBuilder()
+                                             .add("message", exception.getMessage()))
+                                .build())
+                .build();
+    }
+}

